controller/tenant_parallel_test.go

```diff
@@ -0,0 +1,184 @@
+package controller_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/fabric8-services/fabric8-tenant/app"
+	apptest "github.com/fabric8-services/fabric8-tenant/app/test"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/fabric8-services/fabric8-tenant/controller"
+	"github.com/fabric8-services/fabric8-tenant/environment"
+	"github.com/fabric8-services/fabric8-tenant/tenant"
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/assertion"
+	"github.com/fabric8-services/fabric8-tenant/test/doubles"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	tf "github.com/fabric8-services/fabric8-tenant/test/testfixture"
+	"github.com/goadesign/goa"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/h2non/gock.v1"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sync"
+	"testing"
+)
+
+type TenantControllerParallelTestSuite struct {
+	gormsupport.DBTestSuite
+}
+
+func TestTenantControllerInParallel(t *testing.T) {
+	suite.Run(t, &TenantControllerParallelTestSuite{DBTestSuite: gormsupport.NewDBTestSuite("../config.yaml")})
+}
+
+type whenInParallel func(ctx context.Context, service *goa.Service, ctrl app.TenantController)
+
+func (s *TenantControllerParallelTestSuite) TestShowTenantOKWhenNoTenantExistsInParallelForOneUser() {
+	s.verifyTenantCreationWhenNoTenantExistsInParallelForOneUser(
```

Comment:
this is a long method name 😂

Reply:
I know, it's just a replacement of the test and it explains what it does... Do you want me to change it? Proposal?

Reply:
nope, it's ok. The name is self-descriptive ;)